lib/cocina/models/validators/map_coordinates_validator.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+require 'geo/coord'
+
+module Cocina
+  module Models
+    module Validators
+      # Validates that Purl matches the external identifier (druid)
+      class MapCoordinatesValidator
+        def self.validate(clazz, attributes)
+          new(clazz, attributes).validate
+        end
+
+        def initialize(clazz, attributes)
+          @clazz = clazz
+          @attributes = attributes
+        end
+
+        def validate
+          return unless meets_preconditions?
+
+          # valid condition
+          return if invalid_coordinates.empty?
+
+          raise ValidationError, "Invalid map coordinates for #{druid}."
```

Comment:
if this message gets shown in the Argo UI, it might be worth including the invalid coordinates themselves so you can see what the problem is